.github/workflows/build-oci.yaml

```diff
@@ -7,12 +7,25 @@ on:
   pull_request:
     branches:
     - main
-   
+
 env:
   REGISTRY: ghcr.io
   IMAGE_NAME: ${{ github.repository }}
 
 jobs:
+  Lint:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: /home/runner/work/open-feature/flagd
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v1
+        with:
+          go-version: '^1.16.0'
+      - run: make generate
+      - run: make lint
```

Comment:
I think I would rather see this in a new workflow file. In my other PR, I've added a testing phase too, so I will include it here as well.

I think we should keep this file mostly about building the image.

Reply:
I was confused by the name of the workflow `ci` that's why I have put it in the same file.
I will rework this PR to create a new workflow or to add in the testing one tomorrow.

Reply:
> I was confused by the name of the workflow ci that's why I have put it in the same file.

I think it's "oci" as in "OCI Image Format"

> I will rework this PR to create a new workflow or to add in the testing one tomorrow.

:+1: Yep, if you just add the lint stage, I will add testing with my PR after rebasing on yours.

Reply:
> I think it's "oci" as in "OCI Image Format"

What confused me was this line, maybe we should change this to a different name at some point.
https://github.com/open-feature/flagd/blob/317c60a84c2e9168e08442948f75ff849f6eb829/.github/workflows/build-oci.yaml#L1